incremental-font-transfer/src/patchmap.rs

```diff
@@ -734,12 +734,26 @@ impl Ord for IntersectionInfo {
     }
 }
 
+/// Indicates a malformed URI template was encountered.
+///
+/// More info: <https://datatracker.ietf.org/doc/html/rfc6570#section-3>
+#[derive(Debug, PartialEq, Eq)]
+pub struct UriTemplateError;
+
+impl std::fmt::Display for UriTemplateError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "Invalid URI template encountered.")
+    }
+}
+
+impl std::error::Error for UriTemplateError {}
+
 impl PatchUri {
     const BASE32HEX_NO_PADDING: data_encoding::Encoding = new_encoding! {
         symbols: "0123456789ABCDEFGHIJKLMNOPQRSTUV",
     };
 
-    pub fn uri_string(&self) -> String {
+    pub fn uri_string(&self) -> Result<String, UriTemplateError> {
```

Comment:
would it make sense to do our error checking at construction time?

Reply:
We can't determine errors until the template is parsed and for performance reasons we want to avoid parsing until we actually need a concrete URL (since only a small number of the patch URI's will be instantiated in typical cases).

If we switch to sharing the template between multiple URI's then we could pre parse, but that will require larger changes to how things are setup.